File: 조재휘/25-06-2020/소수_찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    bool is_prime;
    vector<int> prime;
    prime.push_back(2);
    for (int i = 3; i <= n; ++i) {
        is_prime = true;
        for (auto iter = prime.begin(); ((*iter) * (*iter)) <= i; ++iter) {
            if ((i % (*iter)) == 0) {
                is_prime = false;
                break;
            }
        }
        if (is_prime) {
            prime.push_back(i);
        }
    }
    answer = prime.size();
    return answer;
}

File: 윤석주/day6~7/소수 찾기.py
import itertools
import math

def solution(numbers):
    answer = 0
    if not len(numbers):
        return answer
    primes = []
    for l in range(len(numbers)):
        p = itertools.permutations(numbers,l+1)
        for number in set(p):
            if isPrime(number,primes):
                answer+=1
    return answer

def isPrime(numbers,primes):
    flag = True
    number = int(''.join(numbers))
    if number < 2:
        return False
    for i in range(2,int(math.sqrt(number)+1)):
        if number % i == 0:
            flag = False
            break
    if flag:
        if number in primes:
            flag = False
        else:
            primes.append(number)
    return flag